### PR TITLE
Add flexpret docs

### PR DESCRIPTION
**Pull request by magnmaeh**
A draft ready in the case that 

https://github.com/lf-lang/reactor-c/pull/412
https://github.com/lf-lang/lingua-franca/pull/2262

get merged. With this PR we should also create a template repository like https://github.com/lf-lang/lf-west-template, just for FlexPRET.

I don't know much about webpages so not quite sure how it's done with the versioned docs part.

### CODE REVIEW

**Comment by magnmaeh** [comment]
@lhstrh Do you think it would be possible to create a lf-flexpret-template repository under the lf-lang organization and give me access to it so I can start developing that one as well? :) If that's something you're allowed to do

**Comment by lhstrh** [comment]
Great idea! The best way to do this is probably to create it under your own account. Then, when it is ready to use, simply transfer the repo to the lf-lang org, and it will become available there.

**Comment by magnmaeh** [comment]
Ah I didn't know that was possible! I'll do that then, thanks! :)

**Comment by lhstrh** [comment]
Yes, it's super easy. Thanks for all the great work you're doing!! 🚀

**Comment by magnmaeh** [comment]
No problem, I'm happy to contribute to such a cool project 😀

**Comment by magnmaeh** [comment]
@lsk567 Do you think it would make sense to repurpose https://github.com/icyphy/lf-flexpret into a template repository, like https://github.com/lf-lang/lf-west-template? Then we would have to transfer ownership too, though. Or is it better to just start from a fresh repo maybe?

**Comment by lsk567** [comment]
> @lsk567 Do you think it would make sense to repurpose https://github.com/icyphy/lf-flexpret into a template repository, like https://github.com/lf-lang/lf-west-template? Then we would have to transfer ownership too, though. Or is it better to just start from a fresh repo maybe?

@magnmaeh Sorry for seeing this message late, Magnus. Yes, let's do it as you suggested. Feel free to reorganize lf-flexpret as you see fit.

**Comment by magnmaeh** [comment]
> > @lsk567 Do you think it would make sense to repurpose https://github.com/icyphy/lf-flexpret into a template repository, like https://github.com/lf-lang/lf-west-template? Then we would have to transfer ownership too, though. Or is it better to just start from a fresh repo maybe?
> 
> @magnmaeh Sorry for seeing this message late, Magnus. Yes, let's do it as you suggested. Feel free to reorganize lf-flexpret as you see fit.

No problem, we decided to just create a new one because the lf-flexpret repo had some Arduino stuff as well :)

**Comment by erlingrj** [comment]
Will review tomorrow

File: docs/embedded/flexpret.mdx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+```
+git clone --recurse-submodules https://github.com/lf-lang/lf-flexpret-template lf-flexpret-workspace && cd lf-flexpret-workspace
+```
+2. Source the `env.bash` or `env.fish` to set up necessary environment variables. You will need to do this every time, so it could be a good idea to add this to your `~/.bashrc`.
+```
+source env.bash
+```
+```
+source env.fish
+```
```
Comment (erlingrj) [inline docs/embedded/flexpret.mdx:26]:
```suggestion
2. Source the `env.bash` to set up necessary environment variables. You will need to do this every time, so it could be a good idea to add this to your `~/.bashrc`.
```
source env.bash
```
```
Dont worry about the fish stuff, any fish user will understand

File: docs/embedded/flexpret.mdx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+## FlexPRET build
+
+3. Now we will build the FlexPRET emulator. Step into the FlexPRET directory and build the default configuration with `cmake`.
+```
+cd $FP_PATH
+cmake -B build && cd build && make
+```
+
+(It is possible to name the build folder something other than `build`, but `bin` is reserved.)
```
Comment (erlingrj) [inline docs/embedded/flexpret.mdx:36]:
```suggestion
```
I think this is unnecessary information

File: docs/embedded/flexpret.mdx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+4. You should now install the FlexPRET build to the FlexPRET software development kit (SDK). This provides the SDK with the necessary knowledge of FlexPRET's hardware configuration, such as the amount of data memory available.
```
Comment (erlingrj) [inline docs/embedded/flexpret.mdx:38]: Could you rephrase "install the FlexPRET build to the FlexPRET software development kit"

**Review by erlingrj — APPROVED** [review]
Great docs! Just a few nitpicks

File: docs/embedded/flexpret.mdx (hunk 3 above)
Comment (magnmaeh) [reply docs/embedded/flexpret.mdx]: Ok, I think it's better now :)
